File: 0160-intersection-of-two-linked-lists/solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    int length(ListNode *headA){
        ListNode* temp=headA;
        int c=0;
        while(temp!=NULL){
            c++;
            temp=temp->next;
        }
        return c;
    }
    ListNode *getIntersectionNode(ListNode *headA, ListNode *headB) {
        int l1=length(headA);
        int l2=length(headB);
        while(l1>l2){
            headA=headA->next;
            l1--;
        }
        while(l2>l1){
            headB=headB->next;
            l2--;
        }
        while(headA!=headB){
            headA=headA->next;
            headB=headB->next;
        }
        return headA;
    }
};

/*
    vector<ListNode *> m;
        struct ListNode* temp=headA;
        while(temp!=NULL){
            m.push_back(temp);
        }
        temp = headB;
        while()
*/

File: 0160-intersection-of-two-linked-lists/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
       int ac=0;
       int bc=0;
       ListNode a= headA;
       ListNode b=headB;

        while(a!=null){
            ac++;
            a=a.next;
        }
        while(b!=null){
            bc++;
            b=b.next;
        }
        while(ac>bc){
            ac--;
            headA=headA.next;
        }
        while(bc>ac){
            bc--;
            headB=headB.next;
        }
        while(headA!=headB){
            headA=headA.next;
            headB=headB.next;
        }
        return headA;
    }
}

/*
 List<ListNode> list = new ArrayList<>();
        ListNode cur1= headA;
        while(cur1!=null){
            list.add(cur1);
            cur1=cur1.next;
        }
        ListNode cur2=headB;
        while(cur2!=null){
            if(list.contains(cur2)){
                return cur2;
            }
            cur2=cur2.next;
        }
        return null;
*/
